test/e2e/network/service.go

```diff
@@ -2027,6 +2027,225 @@ var _ = common.SIGDescribe("Services", func() {
+
+		webserverPod1 := e2epod.NewAgnhostPod(ns, "echo-hostname-1", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod1.Labels = jig.Labels
+		webserverPod1.Spec.NodeName = nodes.Items[1].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod1.Name, f.Namespace.Name, framework.PodStartTimeout))
+		defer func() {
+			framework.Logf("Cleaning up the echo server pod")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod0.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod0.Name)
+			err = cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod1.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod1.Name)
+		}()
+
+		validateEndpointsPortsOrFail(cs, ns, serviceName, portsByPodName{
+			webserverPod0.Name: {servicePort},
+			webserverPod1.Name: {servicePort},
+		})
+
+		ginkgo.By("Creating 2 pause pods that will connect to the webservers")
+		pausePod0 := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pause-pod-0",
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodes.Items[0].Name,
+				Containers: []v1.Container{
+					e2epod.NewAgnhostContainer("agnhost-pause", nil, nil, "pause"),
+				},
+			},
+		}
+		pausePod0, err = cs.CoreV1().Pods(ns).Create(context.TODO(), pausePod0, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, pausePod0.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		pausePod1 := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pause-pod-1",
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodes.Items[1].Name,
+				Containers: []v1.Container{
+					e2epod.NewAgnhostContainer("agnhost-pause", nil, nil, "pause"),
+				},
+			},
+		}
+		pausePod1, err = cs.CoreV1().Pods(ns).Create(context.TODO(), pausePod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, pausePod1.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		defer func() {
+			framework.Logf("Deleting pause pods")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), "pause-pod-0", metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", "pause-pod")
+			err = cs.CoreV1().Pods(ns).Delete(context.TODO(), "pause-pod-1", metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", "pause-pod")
+		}()
+
+		serviceAddress := net.JoinHostPort(serviceIP, strconv.Itoa(servicePort))
+		// assert the target pod for a Service is always going to a local endpoint by checking
+		// the target hostname 3 times.
+		for i := 0; i < 5; i++ {
```

Comment:
the comment doesn't match the loop, right?

Reply:
yeah, will fix that